File: Data Structures & Algorithms/top-k-elements-in-list/submission-2.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int, int> map;
        for (int num : nums) map[num]++;

        struct Compare {
            bool operator()(const std::pair<int,int>& a,
                    const std::pair<int,int>& b) const {
                return a.second < b.second;   // min-heap by second
            }
        };

        priority_queue<pair<int, int>, std::vector<pair<int, int>>, Compare> pq;
        // num, freq
        for (auto it = map.begin(); it != map.end(); it++) {
            pq.push(make_pair(it->first, it->second));
        }

        vector<int> ans;
        for (int i = 0; i < k; i++) {
            pair<int, int> front = pq.top();
            pq.pop();
            ans.push_back(front.first);
        }

        return ans;
    }
};

File: Data Structures & Algorithms/maximum-frequency-after-subarray-operation/submission-2.java
class Solution {
    public int maxFrequency(int[] nums, int k) {
    

        int n = nums.length;
        int max = 0;
        for (int target = 1; target <= 50; target++) {
            if (target == k) continue;
            int count = 0;
            int best = 0;
            for (int num : nums) {
                if (num == target) count++;
                if (num == k) count--;
                count = Math.max(count, 0);
                best = Math.max(best, count);
            }
            max = Math.max(max, best);
        }

        int freq = 0;
        for (int num : nums) {
            if (num == k) freq++;
        }

        return freq + max;
    }
}